### PR TITLE
fix: decuple brush cursor from chart rendering

### PR DESCRIPTION
**Pull request by markov00**
## Summary
This PR remove the unnecessary style recalculation caused by chancing the cursor style on the document body from the `chart_state`.  This was noticeable on this PR https://github.com/elastic/kibana/pull/36517#pullrequestreview-271472260 where the style reflow calculation tooks up to 75ms.

This PR decouple the chart container from the chart renderer, allowing to set correctly the cursor changing the class withour re-rendering the chart component, as the actual implementation.

The cursor style now depends also on the status of the highlighted geometries. If one geometry is highlighted but no `onElementClick` or `onElementOver` listener is available, the cursor will be `default` if the brush is disabled, or `crosshair` if the brush is Enabled.
The pointer is shown only if we have one between `onElementClick` or `onElementOver` enabled and we are over a geometry

**no brush no click/over**
Aug-21-2019 14-44-00

**only brush**
Aug-21-2019 14-44-17

**brush and click**
Aug-21-2019 14-44-30

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- ~[ ] Any consumer-facing exports were added to `src/index.ts` (and stories only import from `../src` except for test data & storybook)~
- [x] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- ~[ ] Proper documentation or storybook story was added for features that require explanation or tutorials~
- ~[ ] Unit tests were updated or added to match the most common scenarios~
- [x] Each commit follows the [convention](https://github.com/elastic/elastic-charts/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: .playground/playgroud.tsx [hunk 1]
```diff
@@ -59,7 +59,7 @@ function renderChart(
   return (
     <div key={key} className="chart">
       <Chart ref={ref}>
-        <Settings tooltip={{ type: 'vertical' }} debug={false} showLegend={true} onCursorUpdate={onCursorUpdate} />
+        <Settings tooltip={{ type: 'vertical' }} debug={false} showLegend={true} onCursorUpdate={onCursorUpdate} onBrushEnd={console.log}/>
```
Comment (nickofthyme) [inline .playground/playgroud.tsx]: remove?

File: src/chart_types/xy_chart/store/chart_state.ts [hunk 2]
```diff
@@ -469,13 +469,6 @@ export class ChartStore {
     } else {
       this.tooltipData.replace(tooltipValues);
     }
-
-    // TODO move this into the renderer
-    if (oneHighlighted) {
-      document.body.style.cursor = 'pointer';
-    } else {
-      document.body.style.cursor = 'default';
-    }
```
Comment (nickofthyme) [inline src/chart_types/xy_chart/store/chart_state.ts]: Nice 🎉

File: .playground/playgroud.tsx (hunk 1 above)
Comment (markov00) [reply .playground/playgroud.tsx]: done in https://github.com/elastic/elastic-charts/pull/331/commits/bc67539eddb6c84cd070a6a3e57511e020669c52

**Comment by markov00** [comment]
@nickofthyme I've updated the PR description fixing the `pointer` on geometries

**Pull request merged**